pkg/ext-proc/backend/datastore.go

```diff
@@ -9,24 +10,65 @@ import (
 	"k8s.io/klog/v2"
 )
 
+func NewK8sDataStore(options ...K8sDatastoreOption) *K8sDatastore {
+	store := &K8sDatastore{
+		poolMu:      sync.RWMutex{},
+		llmServices: &sync.Map{},
+		pods:        &sync.Map{},
+	}
+	for _, opt := range options {
+		opt(store)
+	}
+	return store
+}
+
 // The datastore is a local cache of relevant data for the given LLMServerPool (currently all pulled from k8s-api)
 type K8sDatastore struct {
-	LLMServerPool *v1alpha1.LLMServerPool
-	LLMServices   *sync.Map
-	Pods          *sync.Map
+	// poolMu is used to synchronize access to the llmServerPool.
+	poolMu        sync.RWMutex
+	llmServerPool *v1alpha1.LLMServerPool
```

Comment:
I don't think I agree with keeping these properties private. This assumes that access will only ever be from the `backend` package and I'm not sure that will be true. This package is already rather a catchall and could use some refactoring.

Reply:
> This assumes that access will only ever be from the backend package and I'm not sure that will be true

Then you can make it public or expose a public method. That's what I am doing here. I have synchronized access to the fields so other packages won't mess up with it. 

The bug is that in main.go, it sets a empty LLMServerPool which later in EndpointSlicerReconcilers uses the default port value 0. Making fields private prevents such bugs.